.github/actions/maven-build-and-tag/action.yml

```diff
@@ -0,0 +1,148 @@
+name: Build and tag Maven project
+description: Build and tag Maven project
+inputs:
+  java-version:
+    description: The Java version to perform the build
+    required: false
+    default: '11'
+  java-distribution:
+    description: The Java distribution to perform the build
+    required: false
+    default: 'temurin'
+  property-to-update:
+    description: Property to update in addition to the version of the pom file
+    required: false
+  extra-maven-opts:
+    description: Extra maven options for the build command line
+    required: false
+  nexus-username:
+    description: Nexus user name
+    required: true
+  nexus-password:
+    description: Nexus password
+    required: true
+  quay-username:
+    description: Quay.io user name
+    required: true
+  quay-password:
+    description: Quay.io password
+    required: true
+  docker-username:
+    description: Docker.io user name
+    required: true
+  docker-password:
+    description: Docker.io password
+    required: true
```

Comment:
maybe those could be optional since it depends on the packages you are building if they need to push to quay or docker or both

you could also skip docker-login steps depending on those inputs empty or not (otherwise they will fail)

Reply:
IIRC, we always need to login to Docker Hub, even when we are pushing images to quay.io, that's because some base images and images used by integration tests are hosted in Docker Hub and without login, we run out of quota to pull those images.

Reply:
yeah for your specific use case it's fine but I am always thinking if there is more room to make this action reusable in different contexts

Reply:
It makes sense to make it as reusable as possible, but I prefer to minimize the amount of work without anticipating too much. The risk is spending time adding logic that will never be used. We can always evolve the implementation to cover new scenarios when they are needed. Does it make sense?